app/models/queen.rb

```diff
@@ -1,3 +1,13 @@
 class Queen < Piece
   # - can move in any direction for unlimited distance
+  def valid_move?(end_coord)
+    # Basic checks
+    return false unless valid_move_direction?(end_coord) && game.valid_coordinate?(end_coord)
+    # Check direction
+    direction = get_move_direction(end_coord)
```

Comment:
We don't really need line 7. the `valid_move_direction?` actually uses this method to check if the direction is valid and ensures it is either `:horizontal`, `:diagonal`, or `:vertical`